Pull requests swapil

swapilpaliwal opened the pull request:
# Contains files for interactive visualizations and related documentation.
- Please try running the Ipython notebook file as code needs notebook environment currently for plotting inline in current demos.

swapilpaliwal committed:
Revert "Moved standup from 22 Feb to standups folder"

This reverts commit 02efcd100b241862a26abaa41e0ff1d85f20b2f8.

swapilpaliwal committed:
Revert "Revert "Moved standup from 22 Feb to standups folder""

This reverts commit 1eb6d2daee39671c502e848fd203df6794c06435.

swapilpaliwal committed:
Interactive vis capability added for small set

Requires some changes in basic plot.py .

swapilpaliwal committed:
changed interactivevislib and included basic_plot.py

added some instruction to get files working

swapilpaliwal committed:
added a working ipynb notebook

This notebook shall work by itself and has directions on how to execute
code

swapilpaliwal committed:
Updated the Interactive_vis_toolkit.ipynb for viewing in github

swapilpaliwal committed:
made changes to Interactive_vis_toolkit.ipynb for better code readability

swapilpaliwal committed:
changed interactivevislib.py for better readability

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "ST_Range = 1\n",
+    "S1_Range = 1\n",
+    "\n",
+    "def change_directory():\n",
```
Why would you need to change the working directory for this project?  Can't any calls to this function be addressed by using relative paths since we have a defined directory structure for the project?

Also, now that we have the functions in data_processing.py there should be no more need for any of the plotting functions to read the actual data files -- all plotting functions will be passed a pandas dataframe with the information already read in (and cleaned).  That probably means revising the functions and their documentation below so they accept dataframes as arguments.

cdf6gc replied:
Blake and I are on macs so C: will not work for us. The path of  ../../HDSAviz_data will work so long as we are all consistent in storing the data outside of our repos

houghb replied:
Also, as I mentioned in my note above - this point will be moot once the data_processing functions are incorporated.

swapilpaliwal replied:
Hi thanks for the updates. There are some bugs when i import the Basic_Plot_Updated.py due to which the bokeh plots don't show up and also there are some changes to code which throw unusual errors. Thus i am keeping the original script from Basic_Plot for now. 

There are naming issues like make_plot has been added  instead of makeplot etc. 
I will make it compatible in coming weeks

This is : To Do List item 3.

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "    print 'current working directory: ' + os.getcwd()\n",
+    "\n",
+    "\n",
+    "def gather_all_analysis_files():\n",
```
This function can probably be simplified (or made obsolete) because my function `get_sa()` returns a dictionary with the analysis names as keys, so `dict.keys()` will give the same information as this function.  Wait to edit it until we all go through the new functions in data_processing.py though.

swapilpaliwal replied:
Hey thanks for the comment ... i will update or actually remove the function if needed during this weeks integration of our codes. I am writing it as a To Do task for now.

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "    analysis_files = gather_all_analysis_files()\n",
+    "    tabs_dictionary = {}\n",
+    "    \n",
+    "    for i,files in zip(range(len(analysis_files)), analysis_files):\n",
```
everything in this for loop is also done in data_processing.py

swapilpaliwal replied:
Thanks for the note. 

Making codes compatible is in my to do list. 
I will remove unnecessary functions like this getting files function since its already there but currently my code wasn't compatible with Basic_Plot_Updated.py and thus i couldn't make this change for now....

To Do List 4:

cdf6gc commented on HDSAviz/basic_plot.py:
```diff
@@ -0,0 +1,122 @@
+"""This is a basic script for plotting an analysis file."""
```
I will reformat my code so you can replace this section with an import statement

swapilpaliwal replied:
Ok i will try updating this to import during our integration task this week.
This is in my TO DO list for now.

cdf6gc replied:
Sounds good. Just a heads up I changed the inputs slightly (the minvalues is a float instead of a numpy array) so be sure to check this before running. Also I have default values set so if you just supply a data frame that should work too.

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "    Tab based plotting of Graphs (Only two tabs for simplicity \n",
+    "    and sanity check experimentation). \n",
+    "    \n",
+    "    \"\"\"\n",
```
I needed to add change_directory() into this function first.  However, once makeplot is updated so that it doesn't require filenames anymore (and instead is passed pandas dataframes directly) then this will not be an issue.

swapilpaliwal replied:
Thanks for the Info. I Have changed the path to relative path and thus it should not be a problem when running the code.

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "    variables in different text files.\n",
+    "    \n",
+    "    Input :\n",
+    "    S1_Range: Minimum value of sensitivity for first order effects * 0.001, \n",
```
Can you change the code so that the S1 and ST_Range variables are scaled from 0-1 when they are passed to the function, then within the function if it is necessary to multiply them by 0.001 you can do that, but this way it avoids the user needing to convert them ahead of time.

swapilpaliwal replied:
I have changed the function and variable names to represent decimal values now and hope this addressed the issue. I will try and come up with better text to show up.

cdf6gc commented on HDSAviz/interactivevislib.py:
```diff
@@ -0,0 +1,186 @@
+    minimum threshold of sensitivity selection
+    """
+    #output_notebook()
+    slider1 = FloatSlider(value = 1, min = 0.1, max = 10,
```
I understand this is just for demo but I would add a label or something to indicate that the value set is multiplied by 0.001 through the other function

cdf6gc replied:
I think we need to come up with a come up a consistent way to refer to the different variables to avoid confusion. (i.e. parameters refers to Tmax and C and factors refers to S1 or S2) We can talk about this in person though

houghb replied:
Agreed - I will start a todo list in slack for all the stuff we want to talk about in person

swapilpaliwal replied:
1. I have changed the function to be sightly more intuitive now. I will try to come up with a better way to represent slider without underscores. 
2. Yes lets discuss the parameters, variables and technical terminology in person.

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "    minimum threshold of sensitivity selection\n",
+    "    \"\"\"\n",
+    "    output_notebook()\n",
+    "    slider1 = FloatSlider(value = 1, min = 0.1, max = 10, \n",
```
I am confused why the max is 10?  The maximum sensitivity index value is 1.  Is it possible to make the slider go from 0-1?

cdf6gc replied:
If you were to make that change, which I agree with if possible, would it be possible to also increase the precision of the step the slider takes like 0.001 instead of 0.1?

houghb replied:
If so, we should look at the impact on interactive performance.  If the slider step size is 0.001 then sliding from 0 to 0.5 will require the computer to compute new values 500 times before you get from where you are going to what you want to see...  This is the main problem with sliders in my opinion, they calculate everything in between your start and end points.

swapilpaliwal replied:
Thanks for the feedback. I have now changed the style of slider to zeroes after decimals. 
It basically plots the function for minimum sensitivity of **1, 0.1, 0.01, 0.001, 0.0001, 0.00001 upto 0.00000001**. Hope this addresses both issues by **not taking too much time to compute many values as it now gets on 9 values and has values as low as in range 0.00000001.**

houghb commented on HDSAviz/Interactive_vis_toolkit.ipynb:
```diff
@@ -0,0 +1,443 @@
+    "def Interact_with_burtin_plots():\n",
+    "    \"\"\"\n",
+    "    This function adds capability to adjust the minimum threshold for \n",
+    "    sensitiity plotting\n",
```
typo: sensitivity

swapilpaliwal replied:
corrected.

houghb commented:
These plots are looking very promising!  Am I correct that the ipython notebook is just a combination of the other two files in your branch?  I made all my comments on the notebook code.

houghb commented:
I haven't run this code through pep8 yet since it is in a notebook, but many of the variable names do not conform to pep8 so will need to be adjusted

cdf6gc commented on HDSAviz/interactivevislib.py:
```diff
@@ -0,0 +1,186 @@
+    print 'all analysis files gathered from current directory!'
+    return analysis_files
+
+def plot_all_outcomes_burtin(S1_Range,ST_Range):
```
Blake mentioned that a save option might would be a nice addition to the plot code. I was wondering if it would be possible or helpful to make that a save button?

houghb replied:
Save button would be even better

swapilpaliwal replied:
I think it can be done using save button in bokeh plots... i sent out some hints regarding hat via import previewsave tool and i also incorporated some hard coded saving options for files. see demo.html and multitab_sensitivity_plots.html

cdf6gc replied:
Yeah I added this to my function so we should be set with a save button.

swapilpaliwal committed:
added recommended changes and some plots

adding demo.html and multitab_sensitivity_plot.html along with
restructuring of variables for slider and option to save plots

swapilpaliwal committed:
Made more changes based on pull request suggestions

Things done:
1. made sliders more intuitive
2. removed typos
3. added the save functionality
4. used relative paths instead of absolute paths
5. checked for pep 8 compatibility

Things still To DO :
1. Integration with Basic_Plot_Updated.py
2. Remove unnecessary functions.
3.  use functions from processing.
4.

swapilpaliwal committed:
updated driver

swapilpaliwal committed:
Interactive plotting function added and some demos
